src/test/resources/com/atomist/rug/test/gherkin/handler/event/PassingFeature1StepsAgainstGeneratedWithArrays.ts

```diff
@@ -0,0 +1,17 @@
+import {Given,When,Then, HandlerScenarioWorld} from "@atomist/rug/test/handler/Core"
+
+import {Push} from "@atomist/rug/cortex/stub/Push"
+import {Commit} from "@atomist/rug/cortex/stub/Commit"
+
+ 
+Given("a sleepy country", f => {
+})
+When("a visionary leader enters", world => {
+   world.registerHandler("ReturnsEmptyPlanEventHandlerGenWithArrays")
+   let p = new Push().addContains(new Commit())
```

Comment:
How could we make these methods prettier or more intuitive? I mean, `push.addContains(commit)` isn't as intuitive as `push.addCommit(commit)`. Personally, I even prefer properties access such as `push.commits.push(commit)`, though this means you can't chain calls obviously so I'm happy with the method wrapper.

So far, the TS interface is generated from the model but I feel we probably should have a TS design story driven by the need for a least surprising, consistent and simple TS interface. We would hide away the plumbing to make it match with the model.